### PR TITLE
[Merged by Bors] - feat: add `star_mul_self_add_self_mul_star`

### PR DESCRIPTION
**Pull request by j-loreaux**
adds the lemma: `star a * a + a * star a = 2 • ((ℜ a) ^ 2 + (ℑ a) ^ 2)` for star algebras over `ℂ`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Complex/Module.lean [hunk 1]
```diff
@@ -582,6 +582,18 @@ lemma Complex.coe_realPart (z : ℂ) : (ℜ z : ℂ) = z.re := calc
   _          = z.re := by
     rw [map_add, AddSubmonoid.coe_add, mul_comm, ← smul_eq_mul, realPart_I_smul]
     simp [conj_ofReal, ← two_mul]
+
+lemma star_mul_self_add_self_mul_star {A : Type*} [Ring A] [StarRing A]
```
Comment (eric-wieser) [inline Mathlib/Data/Complex/Module.lean:586]: Is dropping unitality of A easy here?

Comment (j-loreaux) [reply Mathlib/Data/Complex/Module.lean]: good point, thanks! done.

**Review by eric-wieser — APPROVED** [review]
bors d+

Is non-associativity also easy and/or something you care about?

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**